feat: add `group-priority` plugin

chingor13 opened the pull request:
This plugin allows you to configure pull request by priority. If enabled and if a pull request of a prioritized group is found, release-please will limit the proposed release pull requests to the prioritized group only.

For example, java snapshot PRs are now marked as part of the `snapshot` group. If you configure the `group-priority` plugin with the group set to `['snapshot']`, then release-please will only open pull requests for `snapshot` pull requests if there are any.

Fixes #1657

bcoe commented on src/factories/plugin-factory.ts:
```diff
@@ -81,6 +83,13 @@ const pluginFactories: Record<string, PluginBuilder> = {
       options.repositoryConfig,
       (options.type as SentenceCasePluginConfig).specialWords
     ),
+  'group-priority': options =>
```
Eventual cleanup, it would be nice if we could figure out how t instantiate plugins without needing to enumerate them all in this lookup. Perhaps we could introduce a generated `index.ts` to the plugins folder, which would walk the plugins folder during a build step.

bcoe commented on src/manifest.ts:
```diff
@@ -198,9 +198,13 @@ export interface SentenceCasePluginConfig extends ConfigurablePluginType {
 export type PluginType =
   | DirectPluginType
   | ConfigurablePluginType
+  | GroupPriorityPluginConfig
```
In general, I think it would be worth thinking about an eventual refactor that allows us to add additional plugins, options, etc., updating less lists of types, factories, etc.

On the positive, I like that the manifest + plugin based approach requires less updates to add new features, than adding new command line flags.

chingor13 commented on src/plugins/group-priority.ts:
```diff
@@ -0,0 +1,93 @@
+    groups: string[]
+  ) {
+    super(github, targetBranch, repositoryConfig);
+    this.groups = groups;
```
Added a jsdoc block to the constructor

chingor13 commented:
> I think we should add some documentation that describes the plugin.

Added a description/example to the markdown docs with the existing plugin documentation.
Fixed the description in the config schema.
Added jsdoc to the plugin constructor.
